### ISSUE TITLE
No Video in step 05

### ISSUE THREAD

**Issue by ptifranfran** [googlecodelabs/webrtc-web#9]
Hi,

I got through all the steps with caution, my node server is running, my webconsole is positive about people coming in the chatroom and about the connections being made but there isn't any display of video on any tab ! 

![15](https://cloud.githubusercontent.com/assets/24276626/20763131/1d01fafa-b729-11e6-9270-d14d7eff52ff.png)

![16](https://cloud.githubusercontent.com/assets/24276626/20763241/6fdc1166-b729-11e6-83f3-f5a78460caab.png)

![17](https://cloud.githubusercontent.com/assets/24276626/20763248/775d1fe8-b729-11e6-99b5-9aae280c64ad.png)

Could you help me please ? :/

**Comment by thebadscarecrow** [comment]
Hello!
I'm using Firefox & Chrome for this step and.. Yes, at the beginning, video wasn't working. Do you try to run
 `npm cache clean` 
and hard refresh the webpage? It worked for me on Firefox. 
However, I still get and error on Chrome about XMLHttpRequest, but I'm using a VM to run my webserver so..
Hope I was useful.. :)

**Comment by ptifranfran** [comment]
Thank you very much it worked but only on Firefox like you !
That's weird and I'm not using any VM :/ Same error as you

Do you have an explanation on why it's not working ?
Also a bit weird it's not asking me for creating any chatroom ???

**Comment by thebadscarecrow** [comment]
I'm not quite sure, but I think it's because of https on Chrome, but it need to be checked. Like, I mean, now, Chrome wants HTTPS to run videos on WebRTC (or something like that..), so ... :)

Another thing I noticed: you can't open two videos, one on Chrome and one on Firefox at the same time on the same computer. If you run them separately (like only Chrome or only Firefox: it works fine if we forget about HTTPS, and you can run two videos into two different pages on one of your navigators). 

I'll try to set up again the environment (I'm not on my usual computer right now, and I'll tell you about it! ;) )

**Comment by ptifranfran** [comment]
I don't know if that can help you but for my last test, I set up a nginx server on my ubuntu and opened the page on the Firefox of my other computer on windows and the Firefox of the ubuntu on which the server was running and it worked.

I added secure connections to nginx and the first steps I was able to connect in https and see my video with the windows conecting to nginx but on the last steps I got the same error you did :/ 

Weird cause it's working fine with Firefox and the https is up and running

Thank you for your time :)

**Comment by thebadscarecrow** [comment]
Thanks for the help too! I'll try that when I'll have some time. ;)
But yeah, I think I need to understand better how the whole thing works, to fix it better. :)

Thanks for your time too! ;)

**Comment by ptifranfran** [comment]
I think I have an explanation but I might be wrong :/

In order for the call to be successfull, you have to use firefox calling http://your_ip:8080
Why connect to this port number ? Because you're directly connecting to the node server running on your system.

If you're trying from firefox to connect in https directly to this server, it will fail. Because in this exemple, the node server does not seem to be able to cope with https.
The problem being that chrome needs https to accept the videocommunication.

It also explains why it's not asking a room name to connect, cause you're not connecting directly to your web server ! 

So basically this example seems very old and not to be working very well :/

If you have a solution to make this work over https (so with every web browser) it would be really great ! I'm not really good with javascript and nodejs ...

**Comment by nitobuendia** [comment]
That is correct, webRTC requires https when running from a server.
localhost is an exception; which is where this prepared to run in http.

A potential fix is provided in #48. Would this solve this issue, @thebadscarecrow ?

**Comment by samdutton** [comment]
@nitobuendia — will close as per comments, and suggested fix in #48 

Thanks @ptifranfran and @thebadscarecrow — apologies for the delay in responding. Re-open if necessary.

**Comment by thebadscarecrow** [comment]
Hi !
@samdutton : It's okay for the delay, don't worry. :) 
I hadn't much time to check the fix right now, but thanks a lot, @nitobuendia :) Maybe I'll try one day. (and it's seems to work, as you write your in #48 post :) )

Cheers!

**Comment by samdutton** [comment]
Cheers Anaïs!
